packages/select/src/components/multi-select/multiSelect.tsx

```diff
@@ -406,6 +450,26 @@ export class MultiSelect<T> extends AbstractPureComponent<MultiSelectProps<T>, M
         };
     };
 
+    private getKeyDownHandler = (handleQueryListKeyDown: React.KeyboardEventHandler<HTMLElement>) => {
+        return (e: React.KeyboardEvent<HTMLElement>) => {
+            if (this.state.isOpen) {
+                handleQueryListKeyDown?.(e);
+            }
+
+            this.props.popoverTargetProps?.onKeyDown?.(e);
+        };
+    };
+
+    private getKeyUpHandler = (handleQueryListKeyUp: React.KeyboardEventHandler<HTMLElement>) => {
+        return (e: React.KeyboardEvent<HTMLElement>) => {
+            if (this.state.isOpen) {
+                handleQueryListKeyUp?.(e);
+            }
+
+            this.props.popoverTargetProps?.onKeyDown?.(e);
+        };
+    };
```

Comment:
wondering if these should still be wrapping the popover target or the query list? it seems like maybe it should get split up

from the docs app example it seems to be working though so maybe events are bubbling up from the input through the popover? I don't think we'd want to rely on that though